plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -429,6 +432,41 @@ export class AwsLambdaInstrumentation extends InstrumentationBase<AwsLambdaInstr
     return propagation.extract(otelContext.active(), httpHeaders, headerGetter);
   }
 
+  private static _extractOtherEventFields(event: any): Attributes {
+    const answer: Attributes = {};
+    const fullUrl = this._extractFullUrl(event);
+    if (fullUrl) {
+      answer[ATTR_URL_FULL] = fullUrl;
+    }
+    return answer;
+  }
+
+  private static _extractFullUrl(event: any): string | undefined {
+    // API gateway encodes a lot of url information in various places to recompute this
+    if (!event.headers || !event.path) {
+      return undefined;
+    }
+    let answer = '';
+    if (event.headers['x-forwarded-proto']) {
+      answer += event.headers['x-forwarded-proto'] + '://';
+    }
+    if (event.headers['host']) {
+      answer += event.headers['host'];
```

Comment:
I think, if we are unable to resolve host and schema, we might skip extracting full url. Because according to spec, host and schema is mandatory elements of the full url: https://opentelemetry.io/docs/specs/semconv/http/http-spans/